src/main/webapp/app/account/password-reset/init/password-reset-init.component.ts

```diff
@@ -16,55 +16,55 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Component, OnInit, AfterViewInit, Renderer, ElementRef } from '@angular/core';
-import { EMAIL_NOT_FOUND_TYPE } from 'app/shared';
+import { AfterViewInit, Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+import { FormBuilder, Validators } from '@angular/forms';
+import { EMAIL_NOT_FOUND_TYPE } from 'app/shared/constants/error.constants';
+
 import { PasswordResetInitService } from './password-reset-init.service';
-import { PasswordResetService } from 'app/core';
+import { PasswordResetService } from 'app/core/auth/password-reset.service';
 
 @Component({
-    selector: 'jhi-password-reset-init',
-    templateUrl: './password-reset-init.component.html'
+  selector: 'jhi-password-reset-init',
+  templateUrl: './password-reset-init.component.html'
 })
 export class PasswordResetInitComponent implements OnInit, AfterViewInit {
-    error: string;
-    errorEmailNotExists: string;
-    resetAccount: any;
-    success: string;
-    isMailEnabled: boolean;
+  @ViewChild('email', { static: false })
+  email?: ElementRef;
+  errorEmailNotExists: string | undefined;
 
-    constructor(
-        private passwordResetInitService: PasswordResetInitService,
-        private elementRef: ElementRef,
-        private renderer: Renderer,
-        private passwordResetService: PasswordResetService
-    ) {}
+  success = false;
+  isMailEnabled: boolean;
+  resetRequestForm = this.fb.group({
+    email: ['', [Validators.required, Validators.minLength(5), Validators.maxLength(254), Validators.email]]
+  });
 
-    ngOnInit() {
-        this.resetAccount = {};
-        this.isMailEnabled = true;
-        this.passwordResetService.getMailStatus().subscribe(result => (this.isMailEnabled = result['mailEnabled']));
-    }
+  constructor(
+    private passwordResetInitService: PasswordResetInitService,
+    private fb: FormBuilder,
+    private passwordResetService: PasswordResetService
+  ) {
+    this.isMailEnabled = true;
+  }
 
-    ngAfterViewInit() {
-        this.renderer.invokeElementMethod(this.elementRef.nativeElement.querySelector('#email'), 'focus', []);
-    }
+  ngOnInit(): void {
+    this.passwordResetService.getMailStatus().subscribe(result => (this.isMailEnabled = result['mailEnabled']));
```

Comment:
I think this `mailEnabled` property has been removed as per fda468437d781a7f0ab068255495eab7af3fa7a0. I suggest removing this property altogether. Even in the dev profile the mail could be enabled using [MailHog](https://github.com/mailhog/MailHog) as mentioned in the README.md so I don't know why we have to have an option to disable it. :thinking:

Reply:
I think the code is a bit cluttered here. I would suggest creating a separate issue to track the cleanup.

Reply:
Done at, https://github.com/jhipster/jhipster-online/issues/166. Would you be working on this or shall I go ahead? :smile: Since without this change the email resets will not be working, we should probably include this in the upgrade right? :thinking:

Reply:
@SudharakaP, I don't see an issue with password reset (atleast in `dev` profile). Can you elaborate more on the issues you noticed?

Reply:
@vishal423 : That's strange. When I try to access the password reset page it always gives the message, 

![image](https://user-images.githubusercontent.com/12435965/83443054-f757c800-a3fd-11ea-965c-13d67f00ac3f.png)

in both dev or prod.

Reply:
that's the correct screen as you can't send emails with disabled email functionality. That's why you see an option on the user management page to generate a user password reset link.

Reply:
@vishal423 : Yes, but in prod this should be enabled right? So how do we enable it? Or are you implying that even in prod this is what you should see?

Reply:
I think so.. I don't have much background on this decision. That's reason I asked you to create a separate issue to discuss on the expected behavior

Reply:
I did. And it was already merged; https://github.com/jhipster/jhipster-online/issues/166. Feel free to let me know if you see any issues or any suggestions on this matter. :thinking: